src/vm/Components/Assembler/Assembler.java

```diff
@@ -0,0 +1,208 @@
+                            symbolTable.replace(param, address);
+                        } else {
+                            symbolTable.put(param, address);
+                        }
+                    // caso o simbolo ja esteja na tabela de simbolos com um endereco setado
+                    } else if (symbolTable.get(param) != 0) {
+                        // se o simbolo estiver sendo redefinido, joga uma exception
+                        if (param == params[0]) {
+                            throw new AssemblerException("Assembler Exception: Invalid argument (" + param + ") at line " + lineCounter + ". Symbol is already defined.");
+                        }
+                    }
+
+                }
+
+            }
+
+            line = reader.readLine();
+            lineCounter++;
+        }
+
+        // testa se algum simbolo nao foi definido ao final do primeiro passo
+        for (String key : symbolTable.keySet()) {
+            if (symbolTable.get(key) == 0) {
+                throw new AssemblerException("Assembler Exception: Missing arguments at input. Found undefined symbols at the end of the first pass.");
+            }
+        }
+    }
+
+    private void secondPass() throws FileNotFoundException {
+        Reader reader = new Reader("MASMAPRG.ASM");
+        Writer writer = new Writer("source-code.txt");
+        String line = reader.readLine();
+        String data;
+        while (line != null) { 
+            String [] params = line.split(" ");
+            for (String param : params) {
+                // caso seja instrucao
+                if (instructionsMap.containsKey(param)) {
+                    data = instructionsMap.get(param).getBinary();
+                // caso seja simbolo (ignora se estiver na frente)
+                } else if (param != params[0] && symbolTable.containsKey(param)) {
+                    data = Integer.toBinaryString(symbolTable.get(param));
+                // caso seja constante/literal
+                } else if (param.replace("@", "").matches("^\\d+$")) {
+                    data = Long.toBinaryString(Long.valueOf(param));
+                // caso nao deva adicionar nada
+                } else {
+                    data = "";
+                }
+
+                if (data != "") {
+                    data = parseStringToBinarySixteenBits(data);
+                    writer.write(data);
+                    
+                }
+            }
+            line = reader.readLine();
+        }
+    }
+
+    private String parseStringToBinarySixteenBits(String binary) {
+        String fullStr;
+        String zeroes = "";
+
+        // completa a string com 0s ate fechar 16
+        for (int i = 0; i < 16 - binary.length(); i++) {
+            zeroes += "0";
+        }
+
+        fullStr = zeroes + binary;
+        return fullStr;
+    }
+
+    private void setInstructionData() {
+        // key = str instruction
+        // value = str binary, int instructionSize
```

Comment:
acho que pode tirar esse comentários aqui, se não vamos usar

Reply:
na vdd esse comentario era mais como uma instrução de como usar os valores que vem do wrapper caso fosse haver alguma modificação, porque não sabia se todas as IDEs que os outros integrantes usam tem a ferramenta de visualizar a origem e os tipos de valores que os parametros do InstructionWrapper recebe. Retiro mesmo assim?

Reply:
resolved in commit 8444da9a8c15349d67279f65b4540eba5a0db00d